tests/test_hubspot_pagination_test.py

```diff
@@ -1,59 +1,102 @@
+        set_up_start = time.perf_counter()
+
+        self.my_timestamp = self.get_properties()['start_date']
+        test_client = TestClient(self.my_timestamp)
+        existing_records = dict()
+        streams = self.expected_streams() - {'email_events'} # we get this for free with subscription_changes
+        limits = self.expected_page_limits()
+         # 'contacts_by_company' stream needs to get companyIds first so putting the stream last in the list
+        stream_to_run_last = 'contacts_by_company'
+        if stream_to_run_last in streams:
+            streams.remove(stream_to_run_last)
+            streams = list(streams)
+            streams.append(stream_to_run_last)
+
+        for stream in streams:
+            # Get all records
+            if stream == 'contacts_by_company':
+                company_ids = [company['companyId'] for company in existing_records['companies']]
+                existing_records[stream] = test_client.read(stream, parent_ids=company_ids)
+            elif stream in {'companies', 'contact_lists', 'subscription_changes', 'engagements'}:
+                existing_records[stream] = test_client.read(stream, since=self.my_timestamp)
+            else:
+                existing_records[stream] = test_client.read(stream)
+                # existing_records['subscription_changes'] = test_client.get_subscription_changes()  # see BUG_TDL-14938
+            # check if we exceed the limit
+            under_target = limits[stream] + 1 - len(existing_records[stream])
+            print(f'under_target = {under_target} for {stream}')
+
+            if under_target >= 0 :
+                print(f"need to make {under_target} records for {stream} stream")
+                #subscription changes will pass in records and under_target to get to the large limit quicker
+                if stream == "subscription_changes":
+                    test_client.create(stream, subscriptions=existing_records[stream], times=under_target)
+                else:
+                    for i in range(under_target):
+                        # create records to exceed limit
+                        if stream == 'contacts_by_company':
+                            # company_ids = [record["company-id"] for record in existing_records[stream]]
+                            test_client.create(stream, company_ids)
+                        else:
+                            test_client.create(stream)
+
+        set_up_end = time.perf_counter()
+        print(f"Test Client took about {str(set_up_end-set_up_start).split('.')[0]} seconds")
+
     def expected_streams(self):
         """
-        All streams are under test
+        All streams with limits are under test
         """
-        return set(self.expected_page_size().keys()).difference({
-            'subscription_changes',
-            'email_events',
-        })
+        streams_to_test =  set(stream for stream, limit in self.expected_page_limits().items() if limit != set())
```

Comment:
this doesn't really matter, but it would also work if you only did ` if limit` since `bool(set())` is False.